src/traversals/metric_travelling_salesman.jl

```diff
@@ -0,0 +1,73 @@
+"""
+	dfs_short_circuit(g, s)
+
+`g` is a spanning tree of some graph.
+Converts the spanning tree into a tour starting at `s`by
+performing depth first search on it but avoids traversing a node more than once.
+The tour is represented by a permutation of the vertices of the graph. 
+"""
+function dfs_short_circuit(g::AbstractGraph, s::T) where T <: Integer
+
+	nvg = nv(g)
+    seen = zeros(Bool, nvg)
+    S = Vector{T}()
+    tour = Vector{T}()
+    sizehint!(S, nvg)
+    sizehint!(tour, nvg)
+    push!(S, s)
+    push!(tour, s)
+    seen[s] = true
+
+    @inbounds while !isempty(S)
+        v = S[end]
+        u = zero(T)
+        for n in outneighbors(g, v)
+            if !seen[n]
+                u = n
+                break
+            end
+        end
+        if u == zero(T)
+            pop!(S)
+        else
+            seen[u] = true
+            push!(S, u)
+            push!(tour, u)
+        end
+    end
+    return tour
+end
+
+"""
+	metric_travelling_salesman(g, s)
+
+`g` is a complete undirected graph.
```

Comment:
use a traitfn if you need to enforce undirectedness.

Reply:
Ok but [Traits.jl](https://github.com/mauro3/Traits.jl) is abandoned

Reply:
We use SimpleTraits which is still active.